docs/developer/extending/plugins.mdx

```diff
@@ -193,3 +195,134 @@ Plugin as a response needs to return the object of type inherited from `django.h
+    Saleor ->>+ OpenIDConnectPlugin: call external_authentication
+    OpenIDConnectPlugin -->>- Saleor: return redirect url to <br>Oauth provider
+    Saleor -->>- Web: Response from plugin
+    Web ->>+ OAuthProvider: Redirect to authorizantion page
+    OAuthProvider -->>- OpenIDConnectPlugin: GET<br> /plugins/<br>mirumee.authentication.openidconnect/<br>callback?state=redirectUrl/&code=YYY
+    OpenIDConnectPlugin ->>+ OAuthProvider: fetch tokens
+    OAuthProvider -->>- OpenIDConnectPlugin: tokens
+    OpenIDConnectPlugin ->> OpenIDConnectPlugin: validate tokens
+    OpenIDConnectPlugin ->> OpenIDConnectPlugin: create internal tokens with granted permissions
+    OpenIDConnectPlugin -->> Web: set cookie refreshToken with <br>refresh token and redirect to<br> http://localhost:3000/?token=ZZZ&refreshToken=WWW&csrfToken=EEE
+`}
+/>
+
+#### External refresh
+<Chart
+  definition={`
+  sequenceDiagram
+    Web ->>+ Saleor: mutatuion ExternalRefresh with<br> refreshToken and csrf token as an input*
+    Saleor ->>+ OpenIDConnectPlugin: call <br>PluginManager.external_refresh
+    OpenIDConnectPlugin ->>OpenIDConnectPlugin: validate csrf and <br>refresh token structure
+    OpenIDConnectPlugin ->>+ OAuthProvider: refresh tokens
+    OAuthProvider -->>- OpenIDConnectPlugin: new tokens
+    OpenIDConnectPlugin ->> OpenIDConnectPlugin: create new internal tokens
+    OpenIDConnectPlugin -->>- Saleor: new tokens
+    Saleor ->>- Web: return access_token<br> and refresh_token
+`}
+/>
+
+> \* refresh token can be also passed as a cookie - refreshToken
+
+#### External verify
+<Chart
+  definition={`
+  sequenceDiagram
+    Web ->>+ Saleor: mutatuion ExternalVerify with<br> token as an input
+    Saleor ->>+ OpenIDConnectPlugin: call <br>PluginManager.external_verify
+    OpenIDConnectPlugin ->>OpenIDConnectPlugin: decode token
+    OpenIDConnectPlugin -->>- Saleor: return assigned user<br> and payload data
+    Saleor ->>- Web: return token data
+`}
+/>
+
+#### External logout
+<Chart
+  definition={`
+  sequenceDiagram
+    Web ->>+ Saleor: mutatuion ExternalLogout with<br> data to attach as an input
+    Saleor ->>+ OpenIDConnectPlugin: call <br>PluginManager.external_logout
+    OpenIDConnectPlugin ->>OpenIDConnectPlugin: prepare logout url:<br> logoutUrl + input data as GET params
+    OpenIDConnectPlugin -->>- Saleor: return logoutData
+    Saleor ->>- Web: return logoutData
+`}
+/>
+
+#### Authenticate user
+<Chart
+  definition={`
+  sequenceDiagram
+    Web ->>+ Saleor: Request to protected resources with token
+    Saleor ->>+ OpenIDConnectPlugin: Call in middleware <br>Plugin.authenticate_user
+    OpenIDConnectPlugin ->> OpenIDConnectPlugin: Decode and validate token
+    OpenIDConnectPlugin ->> OpenIDConnectPlugin: get user
+    OpenIDConnectPlugin ->> OpenIDConnectPlugin: assign permissions <br>from token
+    OpenIDConnectPlugin ->>- Saleor: User authenticated
+    Saleor ->>- Web: requested resources
+`}
+/>
+
+### OAuth permissions
```

Comment:
Can this be disabled? We should provide a complete list of scopes that need to exist for the plugin to work (the request will likely fail if the auth provider does not know any of the scopes requested).

Reply:
It is optional. If you don't have defined permissions scopes in your audience, we will get a standard token payload. In that case, each user has empty permissions.
I will edit this section to be more precise.

Reply:
Are you sure no popular providers will fail the request with "unknown scopes"?

Reply:
Ech. Thanks for catching this. Again, I've focused on the Auth0.